simde/simde-common.h

```diff
@@ -322,6 +322,11 @@
 #  define SIMDE_VECTORIZE_SAFELEN(l) HEDLEY_PRAGMA(clang loop vectorize_width(l))
 #  define SIMDE_VECTORIZE_REDUCTION(r) SIMDE_VECTORIZE
 #  define SIMDE_VECTORIZE_ALIGNED(a)
+#elif defined(__e2k__)
+#  define SIMDE_VECTORIZE
+#  define SIMDE_VECTORIZE_SAFELEN(l)
+#  define SIMDE_VECTORIZE_REDUCTION(r)
+#  define SIMDE_VECTORIZE_ALIGNED(a)
```

Comment:
This also seems like a compiler issue, not architecture.  What goes wrong here?

I guess lcc defines `__GNUC__`/`__GNUC_MINOR__`/`__GNUC_PATCHLEVEL__` but doesn't implement these?  Maybe it would be better to [treat lcc as a separate compiler](https://github.com/nemequ/hedley/blob/4a59eb6d8e3e73b6c60aed2c92a6590d26df93d8/hedley.h#L461-L479)?  If not,'instead of this how about changing the check on the next line to be `HEDLEY_GCC_VERSION_CHECK(4,9,0) && !defined(__e2k__)` (except preferably not based on the arch).

Reply:
Yes, LCC defines these `__GNUC_*`, but has not the exact same support for the things original GCC has. It's not such far from GCC to be treated as another compiler I think, so I've chose the second way. Done it too.